File: src/main/java/net/gtaun/shoebill/amx/types/ReferenceString.java
package net.gtaun.shoebill.amx.types;

/**
 * Created by marvin on 05.03.15 in project shoebill-api.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
public class ReferenceString {

    private String value;
    private int length;

    public ReferenceString(String value, int length) {
        this.value = value;
        this.length = length;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public int getLength() {
        return length;
    }
}

File: src/main/java/net/gtaun/shoebill/amx/types/ReferenceString.kt
package net.gtaun.shoebill.amx.types

/**
 * This class shall be used with the amx interface when passing a reference string value.
 *
 * @author Marvin Haschker
 */
data class ReferenceString @JvmOverloads constructor(var value: String = "", val length: Int)